GVRf/Framework/framework/src/main/jni/vulkan/vk_render_texture_onscreen.cpp

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "vk_render_texture_onscreen.h"
+#include "vk_render_to_texture.h"
+#include "../engine/renderer/vulkan_renderer.h"
+#include "vk_imagebase.h"
+
+namespace gvr{
+    VkRenderTextureOnScreen::VkRenderTextureOnScreen(int width, int height, int sample_count):VkRenderTexture(width, height, sample_count){
+        initVkData();
+    }
+
+    void VkRenderTextureOnScreen::bind() {
+        if(fbo == nullptr){
+            fbo = new VKFramebuffer(mWidth,mHeight);
+            createRenderPass();
+            VulkanRenderer* vk_renderer= static_cast<VulkanRenderer*>(Renderer::getInstance());
+
+            fbo->createFrameBuffer(vk_renderer->getDevice(), DEPTH_IMAGE | COLOR_IMAGE, mSamples, true);
+        }
+    }
+
+    void VkRenderTextureOnScreen::initVkData(){
```

Comment:
I don't see the reason why initVkData is virtual and has identical impl in both Off and OnScreen.